src/main/webapp/script.js

```diff
@@ -173,6 +173,42 @@ function getGraphDisplay(graphNodes, graphEdges) {
     const node = evt.target;
     node.tip.show();
   });
+
+  let searchElement = document.getElementById('search');
+  searchElement.onchange = function() {
+    if (searchNode(cy, searchElement.value)) {
+      document.getElementById('searcherror').innerText = "";
+    } else {
+      document.getElementById('searcherror').innerText = "Node does not exist.";
+    }
+  };
+  
+}
+
+/**
+ * Zooms in on specific node
+ */
+function searchNode(cy, query) {
+  let target = findNodeInGraph(cy, query);
+  if (target) {
+    cy.fit(target, 50);
```

Comment:
Could you also blur the surrounding nodes and maybe highlight which node is the target node? (because when I tried running it, there were a lot of nodes on the screen and it was kind of difficult to see!)

Reply:
couldn't figure out how to do that with cytoscape but I made it so it changed the color of the node so its more obvious now

Reply:
maybe look at the wine ex that ishani brought up?